blockchain/factory/factory.go

```diff
@@ -0,0 +1,37 @@
+package factory
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/blockchain/endpointProviders"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/blockchain/finalityProvider"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+)
+
+// CreateEndpointProvider creates a new instance of EndpointProvider
+func CreateEndpointProvider(entityType core.RestAPIEntityType) (EndpointProvider, error) {
+	switch entityType {
+	case core.ObserverNode:
+		return endpointProviders.NewNodeEndpointProvider(), nil
+	case core.Proxy:
+		return endpointProviders.NewProxyEndpointProvider(), nil
+	default:
+		return nil, fmt.Errorf("%w: %s", ErrUnknownRestAPIEntityType, entityType)
+	}
+}
+
+// CreateFinalityProvider creates a new instance of FinalityProvider
+func CreateFinalityProvider(entityType core.RestAPIEntityType, proxy proxy, checkFinality bool) (FinalityProvider, error) {
+	if !checkFinality {
+		return finalityProvider.NewDisabledFinalityProvider(), nil
+	}
+
+	switch entityType {
+	case core.ObserverNode:
+		return finalityProvider.NewNodeFinalityProvider(proxy)
```

Comment:
shouldn't the entityType and proxyEndpointProvider type be of the same value? could it be taken from the proxy?
can you have a core.Proxy endpointProvider in the proxy with a nodeFinalityProvider?

Reply:
done